controllers/vspherecluster_controller.go

```diff
@@ -422,6 +423,16 @@ func (r clusterReconciler) reconcileDeploymentZones(ctx *context.ClusterContext)
 	return true, nil
 }
 
+func shouldIncludeZone(zone infrav1.VSphereDeploymentZone, cluster *infrav1.VSphereCluster) bool {
+	deploymentZones, annotationExists := cluster.GetAnnotations()["vsphere.infrastructure.cluster.x-k8s.io/adopt-deploymentzone"]
+	// we have not annotated the cluster yet skip all reconciles
+	if !annotationExists {
+		return false
+	}
+	deploymentZonesList := strings.Split(deploymentZones, ":")
+	return zone.Spec.Server == cluster.Spec.Server && len(deploymentZonesList) > sort.SearchStrings(deploymentZonesList, zone.GetName())
```

Comment:
can u explain a bit what does this do? does it rely on the input data is sorted?